archunit/src/test/java/com/tngtech/archunit/library/freeze/TextFileBasedViolationStoreTest.java

```diff
@@ -119,6 +119,18 @@ public void stores_violations_with_line_breaks() {
         assertThat(violations).as("stored violations").containsExactlyElementsOf(expected);
     }
 
+    @Test
+    public void class_can_be_overridden_and_rule_file_name_strategy_supplied() {
+        class MySubclassOfTextFileBasedViolationStore extends TextFileBasedViolationStore {
```

Comment:
I don't think that's correct.  We have to configure archunit to tell it to use this new implementation, using `freeze.store` property, so that means a subclass of `TextFileBasedViolationStore`.